calico_kubernetes/calico_kubernetes.py

```diff
@@ -257,17 +258,44 @@ def _generate_rules(self):
 
         :return list() rules: the rules to be added to the Profile.
         """
+
+        namespace, ns_tag = self._get_namespace_and_tag(pod)
+
         inbound_rules = [
             {
-                'action': 'allow',
-            },
+                "action": "allow",
+                "src_tag": ns_tag
+            }
         ]
 
         outbound_rules = [
             {
-                'action': 'allow',
-            },
+                "action": "allow"
+            }
         ]
+
+        print('Getting Policy Rules from Annotation of pod %s' % pod)
+
+        annotations = self._get_metadata(pod, 'annotations')
+
+        if 'allowFrom' in annotations.keys():
+            # Remove Default Rule (Allow Namespace)
+            inbound_rules = []
+            rules = json.loads(annotations['allowFrom'])
+            for rule in rules:
+                rule['action'] = 'allow'
+                rule = self._translate_rule(rule, namespace)
+                inbound_rules.append(rule)
+
+        if 'allowTo' in annotations.keys():
```

Comment:
Don't implement the 'allowTo' bits right now, since we haven't defined them in the design.